lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -125,33 +125,36 @@ function dosomething_northstar_update_user($user, $password = null) {
 }
 
 /**
- * Get user profile data from Northstar for the given Drupal ID.
+ * Get user profile data from Northstar for the given ID.
  *
- * @param int $drupal_id Drupal user id.
+ * @param int $id Drupal user id.
+ * @param string $type The index type of the ID
  * @return object
  */
-function dosomething_northstar_get_northstar_user($drupal_id) {
-  $northstar_user_info = cache_get('northstar_user_info_' . $drupal_id, 'cache_dosomething_northstar');
+function dosomething_northstar_get_northstar_user($id, $type = 'drupal_id') {
+  $northstar_user_info = NULL;
```

Comment:
actually no, i think it does need to be here. `cache_get` only runs if that conditional is equal to drupal id. so if i fetch by facebook id, and then do `if $northstar_info)` that seems funky?